lua/neo-tree/sources/common/hijack_cursor.lua

```diff
@@ -0,0 +1,38 @@
+local events = require("neo-tree.events")
+local manager = require("neo-tree.sources.manager")
+local log = require("neo-tree.log")
+
+local M = {}
+
+local setup_for_module = function(module)
+  return function()
+    local state = manager.get_state(module)
+    local winid = state.winid
+    if vim.api.nvim_get_current_win() == winid then
+      local node = state.tree:get_node()
+      log.debug("Cursor moved in tree window, updating cursor pos")
+      local cursor = vim.api.nvim_win_get_cursor(0)
+      local row = cursor[1]
+      local current_line = vim.api.nvim_get_current_line()
+      local startIndex, _ = string.find(current_line, node.name, nil, true)
+      if startIndex then
+        vim.api.nvim_win_set_cursor(0, { row, startIndex - 1 })
+      end
+    end
+  end
+end
+
+---Configures the plugin, should be called before the plugin is used.
+---@param config table Configuration table containing any keys that the user
+---wants to change from the defaults. May be empty to accept default values.
+M.setup = function(config, global_config)
+  local modules = { "filesystem", "git_status", "buffers" }
+  for _, module in ipairs(modules) do
```

Comment:
Is this the proper way to initialize something for all sources?

Reply:
No, because the list of sources is not a static thing and will change according to what extensions a user has. The best place to do it right now would be in this block:

https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8c89efb93b8383666a6f989762c2f610f5f2c930/lua/neo-tree/setup/init.lua#L671-L682

Reply:
Done :+1: